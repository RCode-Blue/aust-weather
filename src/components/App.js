import React, { useState } from "react";
import "../styles/styles.css";

import Dropdown from "./Dropdown";

import cities from "../data/openWeather/cities.json";

function App() {
  const [selectedCity, setSelectedCity] = useState(null);

  const renderSelectedCity = () => {
    console.log(selectedCity);
    if (!selectedCity) {
      return <div></div>;
    }
    return <div className="">{selectedCity.name}</div>;
  };

  // console.log(selectedCity);

  return (
    <div className="App">
      <Dropdown
        cities={cities}
        selectedCity={selectedCity}
        setSelectedCity={setSelectedCity}
      />
      {renderSelectedCity()}
    </div>
  );
}

export default App;
