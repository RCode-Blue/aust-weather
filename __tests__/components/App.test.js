import { configure, mount, shallow } from "enzyme";
import React from "react";
import App from "../../src/components/App";
import "../src/styles/styles.css";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = shallow(<App />);
  expect(toJson(tree)).toMatchSnapshot();
});
