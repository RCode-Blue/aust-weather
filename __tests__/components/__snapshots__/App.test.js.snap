// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App container-fluid p-4"
>
  <div
    className=""
  >
    <p
      className="h5"
    >
      Current weather
    </p>
    <Dropdown
      cities={
        Object {
          "cities": Array [
            Object {
              "coord": Object {
                "lat": -33.867851,
                "lon": 151.207321,
              },
              "country": "AU",
              "id": 2147714,
              "name": "Sydney",
              "state": "",
            },
            Object {
              "coord": Object {
                "lat": -35.283459,
                "lon": 149.128067,
              },
              "country": "AU",
              "id": 2172517,
              "name": "Canberra",
              "state": "",
            },
            Object {
              "coord": Object {
                "lat": -27.467939,
                "lon": 153.028091,
              },
              "country": "AU",
              "id": 2174003,
              "name": "Brisbane",
              "state": "",
            },
            Object {
              "coord": Object {
                "lat": -39.200001,
                "lon": -65.76667,
              },
              "country": "AR",
              "id": 3859521,
              "name": "Darwin",
              "state": "",
            },
            Object {
              "coord": Object {
                "lat": -34.920502,
                "lon": 138.600586,
              },
              "country": "AU",
              "id": 7839644,
              "name": "Adelaide",
              "state": "",
            },
            Object {
              "coord": Object {
                "lat": -31.933331,
                "lon": 115.833328,
              },
              "country": "AU",
              "id": 2063523,
              "name": "Perth",
              "state": "",
            },
            Object {
              "coord": Object {
                "lat": -42.87936,
                "lon": 147.329407,
              },
              "country": "AU",
              "id": 2163355,
              "name": "Hobart",
              "state": "",
            },
          ],
        }
      }
      selectedCity={null}
      setSelectedCity={[Function]}
    />
  </div>
  <div
    className="mb-3"
  />
</div>
`;
